cpp/src/IceGrid/Parser.cpp

```diff
@@ -2573,70 +2573,77 @@ Parser::getInput(char* buf, size_t& result, size_t maxSize)
     else
     {
 #if defined(__APPLE__) || defined(__linux__)
-        const char* prompt = parser->getPrompt();
-        char* line = readline(const_cast<char*>(prompt));
-        if(!line)
+        if (isatty(fileno(stdin)) == 1)
         {
-            result = 0;
-        }
-        else
-        {
-            if(*line)
+            const char* prompt = parser->getPrompt();
+            char* line = readline(const_cast<char*>(prompt));
+            if(!line)
             {
-                add_history(line);
-            }
-
-            result = strlen(line) + 1;
-            if(result > maxSize)
-            {
-                free(line);
-                error("input line too long");
                 result = 0;
             }
             else
             {
-                strcpy(buf, line);
-                strcat(buf, "\n");
-                free(line);
+                if(*line)
+                {
+                    add_history(line);
+                }
+
+                result = strlen(line) + 1;
+                if(result > maxSize)
+                {
+                    free(line);
+                    error("input line too long");
+                    result = 0;
```

Comment:
Should be 1

Reply:
Why 1? I think this is just the number of bytes read and we didn't read anything.

Reply:
My mistake, I thought it was an exit status. 0 is fine.